sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVectorUtils.java

```diff
@@ -125,32 +125,45 @@ public static Map<Integer, Integer> toJavaIntMap(ColumnarMap map) {
   }
 
   private static void appendValue(WritableColumnVector dst, DataType t, Object o) {
```

Comment:
If this function is to handle external types (Row), I think matching logical data type makes more sense, as external types are bound to logical types.

Reply:
OK, I will revert the change of this method, but there maybe 2 problem:

- TimestampType is not handled, will throw `UnsupportedOperationException` directly
- appendValue method not handle MapType yet

seems that another pr is needed to solve these?

Reply:
Yea, this is an existing bug and should be fixed separately.

Reply:
> Yea, this is an existing bug and should be fixed separately.

thanks @cloud-fan,  I will follow up on the remaining bug fix later